cmd/snap/main.go

```diff
@@ -118,7 +118,7 @@ func Parser() *flags.Parser {
 
 		os.Exit(0)
 	}
-	parser := flags.NewParser(&optionsData, flags.HelpFlag|flags.PassDoubleDash)
+	parser := flags.NewParser(&optionsData, flags.HelpFlag|flags.PassDoubleDash|flags.PassAfterNonOption)
```

Comment:
We need logic that counters the changed behavior. In other words, if we have a command that does _not_ want positional parameters, or wants just one positional parameter, but gets many, we'll now see both the bogus positional parameters _and all follow up flags_ as "rest". We must error on that case.

This query will probably clarify the problem:
```
$ grep 'Execute(\[' * -r
cmd_changes.go:func (c *cmdChanges) Execute([]string) error {
cmd_changes.go:func (c *cmdChange) Execute([]string) error {
cmd_create_user.go:func (x *cmdCreateUser) Execute([]string) error {
cmd_find.go:func (x *cmdFind) Execute([]string) error {
cmd_help.go:func (cmd cmdHelp) Execute([]string) error {
cmd_list.go:func (x *cmdList) Execute([]string) error {
cmd_snap_op.go:func (x *cmdRemove) Execute([]string) error {
cmd_snap_op.go:func (x *cmdInstall) Execute([]string) error {
cmd_snap_op.go:func (x *cmdRefresh) Execute([]string) error {
cmd_snap_op.go:func (x *cmdTry) Execute([]string) error {
```

All of those for sure (but perhaps the others too) are ignoring the arguments they get, and they should really be erroring out instead, because they don't _want_ to handle the extra arguments.

I suggest implementing a helper function for that: `err := checkNoArgs(args)` or similar.